includes/Core/Authentication/Authentication.php

```diff
@@ -405,6 +405,26 @@ function ( $user_id ) {
 			add_action( 'googlesitekit_authorize_user', $set_initial_version );
 			add_action( 'googlesitekit_reauthorize_user', $set_initial_version );
 		}
+
+		// Maybe refresh token.
+		$maybe_refresh_token = function() {
+			$token = $this->token->get();
+
+			// Do nothing if the token is not set.
+			if ( empty( $token['created'] ) || empty( $token['expires_in'] ) ) {
+				return;
+			}
+
+			// Do nothing if the token expires in more than 5 minutes.
+			if ( $token['created'] + $token['expires_in'] > time() + 5 * MINUTE_IN_SECONDS ) {
```

Comment:
The AC mentions "30 seconds or less" for this period where as this uses 5 minutes.

This sounds good to me as a more proactive refresh, especially since heartbeat ticks can be a few minutes apart. @felixarntz are you okay with this change?

Reply:
:man_facepalming: I don't know why I set 5 minutes here although we discussed it during working on IB. Sorry about it...

Reply:
Using 5 minutes sounds good to me, per Evan's point.